### PR TITLE
Minor/improve skipping output

### PR DESCRIPTION
**Pull request by mathild3r**
- [x] Tests written and linted [ℹ︎](https://github.com/snyk/general/wiki/Tests)
- [ ] Documentation written [ℹ︎](https://github.com/snyk/general/wiki/Documentation)
- [x] Commit history is tidy [ℹ︎](https://github.com/snyk/general/wiki/Git)

### What this does

Adding some information to why an issue gets skipped. 
I modified the displayErrorForIssue so it also says why.
We were also skipping the issue if the ifUpgradeAvailableOnly was true and the issue have no upgrade available but the INFO message was lost in the logs. It is now added to the summary displayed at the end of a run with the associated reason.

I have also added the creation of a log file format JSON. the key is the function in which the failure or warning happened and the value of the error. It might be easier to look for information that way.

See screen shots below

### Notes for the reviewer

_Instructions on how to run this locally, background context, what to review, questions…_

### More information

- [Link to documentation](https://github.com/snyk-tech-services/jira-tickets-for-new-vulns/wiki/)
### Screenshots

_Visuals that may help the reviewer_
Image
Image

Image

Image
Image

### CODE REVIEW

**Comment by lili2311** [comment]
Please could you update the PR description:
- what is the change, why
- the screenshot does not show much, when will I see the new extra info? Can the screenshot show some more?

File: jira.go [hunk 1]
```diff
@@ -54,12 +54,16 @@ func getJiraTickets(Mf MandatoryFlags, projectID string, customDebug debug) (map
...
 		customDebug.Debug("*** ERROR *** Could not get the Jira issues via Snyk API")
+		message := fmt.Sprintf("Could not get the tickets %s\n", err.Error())
+		writeErrorFile("getJiraTickets", message, customDebug)
 		return nil, errors.New("Could not get the tickets")
 	}
 
 	tickets, err := jsn.NewJson(responseData)
 	if err != nil {
 		customDebug.Debug("*** ERROR *** Could not read Jira issues via Snyk API")
+		message := fmt.Sprintf("Could not read Jira issues via Snyk API %s\n", err.Error())
```
Comment (lili2311) [inline jira.go:65]: I can't see any logs for when we choose to skip creating jira tickets based on passed in options like skip issues that are not fixable? 

Can the messages be added for all issues that exist but we chose to skip making a ticket for it aka it is not an error it is a decision that is right now not visible to users.

File: jira.go [hunk 2]
```diff
@@ -175,8 +187,12 @@ func openJiraTicket(flags flags, projectInfo jsn.Json, vulnForJira interface{},
 
 		if er != nil {
 			if er.Error() == "Failed too many time with 50x errors" {
+				message := fmt.Sprintf("*** ERROR *** Failed too many time with 50x errors %s\n", err.Error())
```
Comment (lili2311) [inline jira.go:190]: `time` => `times`

File: jira.go [hunk 3]
```diff
@@ -196,6 +212,8 @@ func openJiraTicket(flags flags, projectInfo jsn.Json, vulnForJira interface{},
 
 		return responseData, ticketFile, nil, jiraApiUrl
 	}
+	message := fmt.Sprintf("*** ERROR *** Failed to create ticket, vuln ID is empty\n")
```
Comment (lili2311) [inline jira.go:215]: why would `vulnId` be empty, what can cause this scenario?

File: jira.go [hunk 4]
```diff
@@ -246,6 +266,8 @@ func openJiraTickets(flags flags, projectInfo jsn.Json, vulnsForJira map[string]
 					flags.optionalFlags.priorityIsSeverity = false
 					responseDataAggregatedByte, ticket, err, jiraApiUrl = openJiraTicket(flags, projectInfo, vulnForJira, customDebug)
 					if err != nil {
+						message := fmt.Sprintf("*** ERROR *** Issue not created \nERROR:%s", err)
```
Comment (lili2311) [inline jira.go:269]: Can we keep the same language `Jira ticket` instead of issue everywhere?

File: jira.go [hunk 5]
```diff
@@ -278,6 +300,8 @@ func openJiraTickets(flags flags, projectInfo jsn.Json, vulnsForJira map[string]
 	project[projectIdString] = ticketArray
 
 	if fullResponseDataAggregated == "" && !flags.optionalFlags.dryRun {
+		message := fmt.Sprintf("*** ERROR *** Request response from %s is empty\n", flags.mandatoryFlags.endpointAPI)
```
Comment (lili2311) [inline jira.go:303]: `Received empty response from %s` is easier to see the reason, instead of having to look after a long URL

**Comment by lili2311** [comment]
Please could you update the PR description:
- what is the change, why
- the screenshot does not show much, when will I see the new extra info? Can the screenshot show some more?
- The title mentions improved output for `skipped` but no logs mentioned anyhing skipped.

**Review by lili2311 — CHANGES_REQUESTED** [review]
Please see comments

File: jira.go (hunk 3 above)
Comment (mathild3r) [reply jira.go:216]: I am not sure it's possible in the real world but if we can't extract the info from the JSON or it doesn't exist for some reason. It's a safety thing, will move the check to earlier to return faster

**Review by lili2311 — APPROVED** [review]

**Pull request merged**